ansible/main.yml

```
@@ -0,0 +1,110 @@
+---
+- hosts: all
+  vars:
+    app_path: /home/ubuntu/RD-WATCH
+    venv: "{{ app_path }}/venv"
+  tasks:
+  - name: checkout code
+    git:
+      repo: "https://github.com/ResonantGeoData/RD-WATCH.git"
+      dest: "{{ app_path }}"
+      force: yes
+
+  - name: Set environment variables
+    set_fact:
+      watch_environment:
+        DJANGO_CONFIGURATION:  "{{ lookup('env', 'DJANGO_CONFIGURATION') }}"
+        DJANGO_DATABASE_URL: "{{ lookup('env', 'DJANGO_DATABASE_URL') }}"
+        DJANGO_CELERY_BROKER_URL: "{{ lookup('env', 'DJANGO_CELERY_BROKER_URL') }}"
+        DJANGO_MINIO_STORAGE_ENDPOINT: "{{ lookup('env', 'DJANGO_MINIO_STORAGE_ENDPOINT') }}"
+        DJANGO_MINIO_STORAGE_ACCESS_KEY: "{{ lookup('env', 'DJANGO_MINIO_STORAGE_ACCESS_KEY') }}"
+        DJANGO_MINIO_STORAGE_SECRET_KEY: "{{ lookup('env', 'DJANGO_MINIO_STORAGE_SECRET_KEY') }}"
+        DJANGO_STORAGE_BUCKET_NAME: "{{ lookup('env', 'DJANGO_STORAGE_BUCKET_NAME') }}"
+
+  - name: install pip + python, postgres headers 
+    apt: 
+      pkg:
+        - python3-pip
+        - python3-dev
+        - libpq-dev
+      update_cache: yes
+      state: present
+    become: yes
+
+  - name: install virtualenv
+    pip:
+      name: 
+        - virtualenv
+    become: yes
+
+  - name: create virtualenv
+    shell:
+      cmd: /usr/local/bin/virtualenv {{ app_path }}/venv -p python3.8
+
+  - name: install pip requirements
+    shell: 
+      cmd: "{{ venv }}/bin/pip install -e .[dev] --find-links=https://girder.github.io/large_image_wheels"
+      chdir: "{{ app_path }}"
+
+  - name: run django migrations
+    django_manage:
+      command: migrate
+      app_path: "{{ app_path }}"
+      virtualenv: "{{ venv }}"
+    environment: "{{ watch_environment }}"
+
+  - name: create conf.d directory
+    file:
+      path: /etc/conf.d
+      state: directory
+    become: yes
+
+  - name: create celery config
+    template:
+      src: "./celery/celery.j2"
+      dest: /etc/conf.d/celery
+    become: yes
+
+  - name: create celery service
+    template:
+      src: "./celery/celery.service.j2"
+      dest: /etc/systemd/system/celery.service
+    become: yes
+
+  - name: create celery directories
+    file:
+      path: "{{ item }}"
+      state: directory
+      mode: 0777
+    loop:
+      - /var/log/celery
+      - /var/run/celery
+    become: yes
+
+  - name: start celery service
+    systemd:
+      name: celery.service
+      enabled: yes
+      state: started
+      daemon_reload: yes
+    become: yes
+
+  - name: create gunicorn config
+    template:
+      src: "./gunicorn/gunicorn.j2"
+      dest: /etc/conf.d/gunicorn
+    become: yes
+
+  - name: create gunicorn service
+    template:
+      src: "./gunicorn/gunicorn.service.j2"
+      dest: /etc/systemd/system/gunicorn.service
+    become: yes
+
+  - name: start gunicorn service
+    systemd:
+      name: gunicorn.service
+      enabled: yes
+      state: started
```

```suggestion
      state: restart
```
To ensure code changes get picked up by the running webserver.